server/queue/queue-jms/src/main/java/org/apache/james/queue/library/MailQueueManagement.java

```diff
@@ -124,7 +124,13 @@ public long removeWithSender(String address) throws Exception {
 
             MailQueueItemView mView = it.next();
             Mail m = mView.getMail();
-            Optional<ZonedDateTime> nextDelivery = mView.getNextDelivery();
+            long nextDelivery = -1;
+            Optional<ZonedDateTime> nextDeliveryTime = mView.getNextDelivery();
+            // check if zone date time is present
+            if (nextDeliveryTime.isPresent()) {
+                ZonedDateTime nextDeliveryZonedDateTime = nextDeliveryTime.get();
```

Comment:
Calling `isPresent` then `get` is an anti pattern for the usage of Optional.

The same code chunck could be written:

```
long nextDelivery = nextDeliveryTime
    .map(time -> time.toInstant().toEpochMilli())
    .orElse(-1L);
```